src/vs/editor/contrib/multicursor/multicursor.ts

```diff
@@ -379,7 +379,8 @@ export class MultiCursorSession {
 
 		const allSelections = this._editor.getSelections();
 		const lastAddedSelection = allSelections[allSelections.length - 1];
-		const nextMatch = this._editor.getModel().findNextMatch(this.searchText, lastAddedSelection.getEndPosition(), false, this.matchCase, this.wholeWord ? this._editor.getOption(EditorOption.wordSeparators) : null, false);
+		const isRegex = this.findController.getState().isRegex;
+		const nextMatch = this._editor.getModel().findNextMatch(this.searchText, lastAddedSelection.getEndPosition(), isRegex, this.matchCase, this.wholeWord ? this._editor.getOption(EditorOption.wordSeparators) : null, false);
```

Comment:
`isRegex` should be a part of the constructor of `MultiCursorSession` and and assigned in `create()`.
Then `isRegex` can be used as an argument on line 434 for `findPreviousMatch()` inside `_getPreviousMatch()`. Useful for the command **Add Selection To Previous Find Match** (`Ctrl` + `Shift` + `D`).